codex-rs/default.nix

```diff
@@ -1,40 +1,53 @@
-{ pkgs, monorep-deps ? [], ... }:
-let
+{
+  pkgs,
+  monorepo-deps ? [],
+  ...
+}: let
+  lib = pkgs.lib;
+
   env = {
     PKG_CONFIG_PATH = "${pkgs.openssl.dev}/lib/pkgconfig:$PKG_CONFIG_PATH";
   };
-in
-rec {
+in rec {
   package = pkgs.rustPlatform.buildRustPackage {
     inherit env;
     pname = "codex-rs";
     version = "0.1.0";
-    cargoLock.lockFile = ./Cargo.lock;
-    doCheck = false;
     src = ./.;
+
+    cargoLock = {
+      lockFile = ./Cargo.lock;
+      # ratatui is patched via git, so we must whitelist its fixed-output hash
+      outputHashes = {
+        "ratatui-0.29.0" = "sha256-HBvT5c8GsiCxMffNjJGLmHnvG77A6cqEL+1ARurBXho=";
```

Comment:
I'm afraid this cannot be managed by dependabot or any automated system.

Reply:
Yeah its super unfortunate, and according to our mutual friend, o3, this seems like the lesser of evils...

I presume you all plan on getting your changes merged to ratatui upstream, that would definitely clean it up. But I'm also happy to help, this flake is a part of my dotfiles now, so you all can ping me whenever the action breaks, honestly I'd much prefer being pinged than the flake not work :)

Reply:
(Pls excuse the drive-by commenting)

If you do want to maintain this long term, you can write a small script that does something to the effect of `nix-prefetch-url --unpack --type sha256 $URL` and writes it into something like `foo.nix` then do `import ./foo.nix` there. Give or take some datastructure adjustments.

Reply:
ya that seems like a lot of extra work for something where there hasnt been changes in 3 weeks: https://github.com/nornagon/ratatui/tree/nornagon-v0.29.0-patch I'd like to better understand the plans to upstream